next/src/form.ts

```diff
@@ -26,6 +30,39 @@ export interface ValidationResult {
   formErrors?: FormErrors
 }
 
+/**
+ * Remove composition keywords and their indices as well as conditional keywords from the path
+ * @param path - The path to clean
+ * @returns The cleaned path
+ * @example
+ * ```ts
+ * cleanErrorPath(['some_object','allOf', 0, 'then', 'field'])
+ * // ['some_object', 'field']
+ * ```
+ */
+function cleanErrorPath(path: ValidationErrorPath): ValidationErrorPath {
```

Comment:
I think we could use a declarative approach for faster readability. Not sure if I missed any corner case here, so feel free to disregard my suggestion if that's the case 👇 

```ts
function cleanErrorPath(path) {
    // Filter out specific keywords and non-string/non-numeric path segments
    return path.filter(segment => 
        typeof segment === 'string' && 
        !['allOf', 'anyOf', 'oneOf', 'then', 'else'].includes(segment)
    );
}
```

Also can we have tests for this function?